### ISSUE TITLE
Differences between CMake build and autotools build

### ISSUE THREAD

**Issue by tgurr** [libuv/libuv#3009]
`libuv-static.pc` appears to be only installed when using CMake but not when using the autotools build system and the resulting static library is named `libuv_a.a` contrary to `libuv.a` when using autotools.

There are also differences in the installed headers, with autotools I get:

```
/usr/x86_64-pc-linux-gnu/include/uv.h
/usr/x86_64-pc-linux-gnu/include/uv/errno.h
/usr/x86_64-pc-linux-gnu/include/uv/threadpool.h
/usr/x86_64-pc-linux-gnu/include/uv/version.h
/usr/x86_64-pc-linux-gnu/include/uv/unix.h
/usr/x86_64-pc-linux-gnu/include/uv/linux.h
```

While when using CMake I end up with additional headers being installed:

```
/usr/x86_64-pc-linux-gnu/include/uv/win.h
/usr/x86_64-pc-linux-gnu/include/uv/tree.h
/usr/x86_64-pc-linux-gnu/include/uv/sunos.h
/usr/x86_64-pc-linux-gnu/include/uv/stdint-msvc2008.h
/usr/x86_64-pc-linux-gnu/include/uv/posix.h
/usr/x86_64-pc-linux-gnu/include/uv/os390.h
/usr/x86_64-pc-linux-gnu/include/uv/darwin.h
/usr/x86_64-pc-linux-gnu/include/uv/bsd.h
/usr/x86_64-pc-linux-gnu/include/uv/android-ifaddrs.h
/usr/x86_64-pc-linux-gnu/include/uv/aix.h
```

* **Version**: 1.40.0
* **Platform**: Linux localhost 5.6.5 #1 SMP Mon Apr 20 11:54:25 CEST 2020 x86_64 GNU/Linux

**Comment by kasperisager** [comment]
I just hit this as well moving from Autotools to CMake for compatibility reasons. Is there a desire to align the two toolchains? If so, I'd be happy to submit a pull request.

**Comment by vtjnash** [comment]
Yes, we generally want them to behave similarly. There are a couple half-complete cmake PRs that you could work from also

**Comment by kasperisager** [comment]
Does the opinion expressed in https://github.com/libuv/libuv/pull/2374#issuecomment-522240794 still stand? That was one of the differences I encountered.

**Comment by vtjnash** [comment]
I think so, as I don't know any solution. But we could probably change the autotools `--enable-static` build to align it with cmake.

**Comment by kasperisager** [comment]
I don't have an overview of how the CMake build is used, though I know the Autotools build is very much depended on to produce `libuv.{a,dylib,so}` artefacts. That does seem to put us at a bit on an impasse if the particular behaviour of both toolchains is depended on. https://xkcd.com/1172/

**Comment by rossburton** [comment]
With libuv 1.51.0 this is still happening. The static library naming appears to be resolved with CMake's static library also being called `libuv.a` but there are still more headers:
```
  /usr/include/uv/aix.h was added
  /usr/include/uv/bsd.h was added
  /usr/include/uv/darwin.h was added
  /usr/include/uv/os390.h was added
  /usr/include/uv/posix.h was added
  /usr/include/uv/sunos.h was added
  /usr/include/uv/tree.h was added
  /usr/include/uv/win.h was added
```

Looks like the autotools build installs just the headers that match the platform, whereas the CMake build installs all possible headers.

**Comment by bnoordhuis** [comment]
Is there a preferred behavior? I think we can make either build match the other if it's important. That said:

- making cmake match autotools seems like it might break existing users that are cross-compiling
 
- no one complained about the autotools build yet so I'm inclined to leave it alone

**Comment by rossburton** [comment]
The autotools layout seems to make more sense: no point in installing headers that are not relevant. The library itself is built for a specific platform, so the headers should match.

**Comment by bnoordhuis** [comment]
No harm in it either, and as I mentioned, changing it now stands a decent chance of breaking existing users. I'm going to leave it as is.

**Comment by rossburton** [comment]
With the cmake build that ships all headers, how are the eg windows-specific headers being used if libuv was built for linux?

**Comment by bnoordhuis** [comment]
ifdef guards.
